sched/definitions.go

```diff
@@ -17,3 +17,33 @@ type Task struct {
 	Command    []string // Argv to run
 	SnapshotId string
 }
+
+type JobState int
+
+const (
+	// An unambiguous 0-value.
+	UNKNOWN JobState = iota
```

Comment:
I'm not suggesting returning in progress as soon as its placed on the Queue.  My suggestion is to remove UNKNOWN.  

PENDING - on the work queue not scheduled
RUNNING - scheduled has a Start Saga Log Entry
DONE - has a Start Saga Log Entry and an End Saga Log Entry

This eliminates the need for status on the work queue.  

I don't like the proposed alternative, we are implementing an async work model and this goes against this paradigm.   Also doing this would result in a lot of false negatives (failed to enqueue) if connections are broken or take longer than expected which can happen in failure modes.  We should not design the system to make assumptions that aren't based on worst case scenarios.  This will lead to every fragile solutions